src/code/client.ts

```diff
@@ -201,7 +201,7 @@ class CloudFileManagerClient {
         if (allProviders[providerName]) {
           Provider = allProviders[providerName]
           // don't add providers that require configuration if no (or invalid) configuration provided
-          if (Provider.hasValidOptions(providerOptions)) {
+          if (!Provider.hasValidOptions || Provider.hasValidOptions(providerOptions)) {
```

Comment:
Did you test this with all the providers?

Reply:
Yes, all the other providers have an implementation for `hasValidOptions` so they should be unaffected by the change.